### PR TITLE
DOC: file obj for to_csv must be newline=''

### LINKED ISSUE #25048: line terminator issue (again) on windows when writing to_csv()
Another addition to windows line-end saga:
Changes in Pandas 0.24.0, related to https://github.com/pandas-dev/pandas/issues/20353, seem to cause some strange behaviour in this not so uncommom edge case:

#### Problem description
When using `df.to_csv()` to write to an exisitng file handle without explicitly setting `newline=''`  (or `newline='\n'`) in the open statement an extra '\r' is added to the default (on windows) '\r\n' line-end:

```python
data = pd.DataFrame({"string_with_lf": ["a\nbc"], "string_with_crlf": ["a\r\nbc"]})
with open("test2.csv", mode='w') as f:
    data.to_csv(f, index=False)
with open("test2.csv", mode='rb') as f:
    print(f.read())
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.7.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None
pandas: 0.24.0
pytest: 4.0.2
pip: 18.1
setuptools: 40.6.3
Cython: 0.29.2
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: 1.8.2
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 3.0.2
openpyxl: 2.5.12
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.2
lxml.etree: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by chris-b1**
- [x] closes #25048

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -1689,7 +1689,7 @@ The ``Series`` and ``DataFrame`` objects have an instance method ``to_csv`` whic
 allows storing the contents of the object as a comma-separated-values file. The
 function takes a number of arguments. Only the first is required.
 
-* ``path_or_buf``: A string path to the file to write or a StringIO
+* ``path_or_buf``: A string path to the file to write or a file object.  If a file object it must be opened with `newline=''`
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/io.rst]: On all platforms, or only for Windows? (the issue seemed to be about windows, but didn't follow it)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks!

File: doc/source/user_guide/io.rst (hunk 1 above)
Comment (chris-b1) [reply doc/source/user_guide/io.rst]: So in practice, it only matters on Windows, but I think it's safe advice to give on all platforms.  Python's csv docs do the same
https://docs.python.org/3/library/csv.html#csv.writer

**Comment by TomAugspurger** [comment]
Restarted to avoid the DST failures.

Just the sparse failure now, which is fixed on master. Should be OK to merge @jorisvandenbossche .

**Pull request merged**